sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -327,6 +326,32 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
         .getOrElse(Array.empty[String])
     val bucketSpec = Option(ctx.bucketSpec()).map(visitBucketSpec)
 
+    val path = new CaseInsensitiveMap(options).get("path")
+    val storage = CatalogStorageFormat(path, Some(provider), options)
+    val columns = Option(ctx.colTypeList()).toSeq.flatMap(visitCatalogColumns)
```

Comment:
yea definitely. SQL is just one of the API entries, we should find a centre place to do error handling so all APIs can benefit